test/e2e/nft/auto-detect-nft.spec.js

```diff
@@ -0,0 +1,37 @@
+const { strict: assert } = require('assert');
+const { defaultGanacheOptions, withFixtures } = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+const { setupAutoDetectMocking } = require('./mocks');
+
+describe('NFT detection', function () {
+  it('displays NFT media', async function () {
+    const driverOptions = { mock: true };
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withNetworkControllerOnMainnet()
+          .withPreferencesControllerNftDetectionEnabled()
+          .build(),
+        driverOptions,
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test.title,
+        testSpecificMock: setupAutoDetectMocking,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
```

Comment:
```suggestion
                await unlockWallet(driver);
```

Reply:
Updated here: https://github.com/MetaMask/metamask-extension/pull/21520/commits/83cb3446e9a70874f1f8234a6d6958a397405ae6. Also updated all the other NFT tests